### ISSUE TITLE
Regression introduced where individual ShapeElementList instances seem to share the last center_x value

### ISSUE THREAD

**Issue by SirGnip** [pythonarcade/arcade#755]
### Actual behavior:
When there are multiple `ShapeElementList` instances, they all seem to share the same `center_x`.

### Expected behavior:
Each individual `ShapeElementList` should respond to its own independent `center_x` member.

### Steps to reproduce/example code:

```
import arcade

class MyGame(arcade.Window):
    def __init__(self, width, height, title):
        super().__init__(width, height, title)

        self.shapes1 = arcade.ShapeElementList()
        shape1 = arcade.create_ellipse_filled(300, 100, 40, 20, arcade.color.ORANGE, 0)
        self.shapes1.append(shape1)

        self.shapes2 = arcade.ShapeElementList()
        shape2 = arcade.create_rectangle_filled(300, 400, 20, 20, arcade.color.BLUE, 0)
        self.shapes2.append(shape2)

    def on_draw(self):
        arcade.start_render()
        self.shapes1.draw()
        self.shapes2.draw()

    def on_update(self, delta_time):
        self.shapes1.center_x += 2
        self.shapes2.center_x += -2

def main():
    MyGame(600, 600, "bug repro")
    arcade.run()

if __name__ == "__main__":
    main()
```

Running this code, I expect that the orange ellipse would move to the right and the blue square would move to the left.  However, with arcade 2.4.2, both shapes move to the left.  The above snippet works as expected in arcade v 2.2.1. This "transform sharing" behavior applies to other members as well (`center_y`, `angle`, etc.).

This bug is NOT seen in the shapes_buffered.py example because the `Shape.draw()` method explicitly sets the `center_x` & `center_y` members each frame before `draw()` is called on the `ShapeElementList`.

    def draw(self):
        self.shape_list.center_x = self.x
        self.shape_list.center_y = self.y
        # self.shape_list.angle = self.angle
        self.shape_list.draw()

**Comment by einarf** [comment]
Fixed in 42d5a2646d0fbff15b0b00f3ec4651bcca329a0d

When we did some major optimizations earlier this summer we started using pre-loaded glsl programs stored in the context. All ShapeLists are now using a shared program and we must set the `Position` uniform in `draw()` instead.

Shapes and ShapeList still need a lot of work moving more logic into shaders + a better solution is needed for growing the buffer. The `draw_<shape>` functions are the only ones that were properly optimized this way. They were measured to be about ~84 times faster on OS X with Intel HD 4000 (my low end benchmark computer)

**Comment by SirGnip** [comment]
Side question... even though draw_* functions have been made much faster, I'm assuming Shapes, ShapeElementList and create_* functions are still faster?

**Comment by einarf** [comment]
They are not faster currently. It's mostly unchanged. They were never really slow to begin with. Once you have the shape list built they are fairly performant. They mostly need some improvements when it comes to restructuing the internal buffers because bulding them is very slow.

The current related tasks
* Optimize Unbuffered Shapes https://github.com/pythonarcade/arcade/issues/663
* The performance of the `ShapeElementList` https://github.com/pythonarcade/arcade/issues/664

Just pre-compiling the shaders optimized the unbuffered shapes a lot. The completed ones have moved most of the logic into geometry shaders so we generate the geometry on the fly (on gpu) instead of building it with python code (orders of magnitude faster)

**Comment by pvcraven** [comment]
Is this still an issue?

**Comment by einarf** [comment]
I verified this is working as intended now.

**Comment by jacobj45** [comment]
I am on version 2.5.7 and the issue still exists for `angle` property. `center_x` and `center_y` properties work fine.
<br>

**Code to reproduce issue**
```
class MyGame(arcade.Window):
    def __init__(self, width, height, title):
        super().__init__(width, height, title)
        self.shapes1 = arcade.ShapeElementList()
        shape1 = arcade.create_ellipse_filled(0, 0, 40, 20, arcade.color.ORANGE, 0)
        self.shapes1.append(shape1)
        self.shapes1.center_x = width / 4
        self.shapes1.center_y = height / 2

        self.shapes2 = arcade.ShapeElementList()
        shape2 = arcade.create_rectangle_filled(0, 0, 20, 20, arcade.color.BLUE, 0)
        self.shapes2.append(shape2)
        self.shapes2.center_x = 3 * width / 4
        self.shapes2.center_y = height / 2

    def on_draw(self):
        arcade.start_render()
        self.shapes1.draw()
        self.shapes2.draw()

    def on_update(self, delta_time):
        self.shapes1.angle += 2
        self.shapes2.angle += -2
```

**Comment by pvcraven** [comment]
This occurs in both dev and maintenance branches. Jacobj45's example expanded a bit:

``` python
import arcade


class MyGame(arcade.Window):
    def __init__(self, width, height, title):
        super().__init__(width, height, title)
        self.shapes1 = arcade.ShapeElementList()
        shape1 = arcade.create_ellipse_filled(0, 0, 40, 20, arcade.color.ORANGE, 0)
        self.shapes1.append(shape1)
        self.shapes1.center_x = width / 4
        self.shapes1.center_y = height / 2

        self.shapes2 = arcade.ShapeElementList()
        shape2 = arcade.create_rectangle_filled(0, 0, 20, 20, arcade.color.BLUE, 0)
        self.shapes2.append(shape2)
        self.shapes2.center_x = 3 * width / 4
        self.shapes2.center_y = height / 2

    def on_draw(self):
        arcade.start_render()
        self.shapes1.draw()
        self.shapes2.draw()

    def on_update(self, delta_time):
        # THIS ROTATES BOTH LISTS
        self.shapes1.angle += 2


def main():
    """ Main method """
    window = MyGame(800, 600, "Hi")
    arcade.run()


if __name__ == "__main__":
    main()```

**Comment by einarf** [comment]
Oh. This is probably something stupid I did (again) 😄

**Comment by Cleptomania** [comment]
Released with 2.6.0
